src/plugins/stacked-barchart/src/StackedBarchartPanel.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useLayoutEffect, useMemo } from 'react';
+import { useMeasure } from 'react-use';
+import { DataFrame, PanelProps, classicColors} from '@grafana/data';
+import { useTheme2 } from '@grafana/ui';
+import * as d3 from 'd3';
+import { css } from 'emotion';
+
+import { StackedBarchartOptions } from 'types';
+
+interface Props extends PanelProps<StackedBarchartOptions> { }
+
+interface IPreparedGraphFields {
+  stacks: Object[],
+  xDistinctGroups: string[],
+  yDistinctGroups: string[],
+  xMaxGroupCount: number,
+  error: string
+}
+
+function prepareGraphableFields(series: DataFrame[] | undefined): IPreparedGraphFields {
+  const result: IPreparedGraphFields = {
+    stacks: [],
+    xDistinctGroups: [],
+    yDistinctGroups: [],
+    xMaxGroupCount: 0,
+    error: ''
+  }
+  if (!series?.length) {
+    result.error = 'No data';
+    return result;
+  }
+  if (series[0]?.fields?.length !== 3) {
+    result.error = 'Data should have 3 columns';
+    return result;
+  }
+  const xGroup = series[0].fields[0].values.toArray();
+  const yGroup = series[0].fields[1].values.toArray();
+  const yCount = series[0].fields[2].values.toArray();
+  let xGroupCount = 0;
+  const xGroupByYgroup = xGroup.reduce(function(storage, item, index) {
+    const last = storage[storage.length - 1];
+    if(last?.group == item){
+      last[yGroup[index]] = yCount[index];
+      xGroupCount += yCount[index];
+    } else {
+      storage.push({"group": item, [yGroup[index]]: yCount[index]});
+      result.xDistinctGroups.push(item);
+      if (result.xMaxGroupCount < xGroupCount) {
+        result.xMaxGroupCount = xGroupCount;
+      }
+      xGroupCount = yCount[index];
+    }
+    return storage;
+  }, []);
+  if (result.xMaxGroupCount < xGroupCount) {
+    result.xMaxGroupCount = xGroupCount;
+  }
+  result.yDistinctGroups = Object.keys(xGroupByYgroup[0]).filter((x) => x != "group");
+  result.stacks = d3.stack().keys(result.yDistinctGroups)(xGroupByYgroup);
+  return result;
+}
+
+function countDigits(value: number): number {
+  return 1 + Math.floor(Math.log(value) / Math.log(10));
+}
+
+export const StackedBarchartPanel: React.FC<Props> = ({ options, data, width, height }) => {
+
+  const graphRef = React.useRef(null);
+  const graphWrapperRef = React.useRef(null);
+  const theme = useTheme2();
```

Comment:
What is the difference between `useTheme` and `useTheme2`? Why not using `getTheme`?

Reply:
It is quite confusing indeed; `useTheme` is deprecated, `useTheme2` is in the beta stage, `getTheme` creates a new theme instead of retrieving the current theme. Updated this part to use `useTheme`.